src/consensus/validator.rs

```diff
@@ -131,6 +135,12 @@ impl ShardValidator {
         } else {
             panic!("No proposer set");
         }
+        if let Err(err) = self
+            .proposal_store
+            .delete_proposal(commits.height.unwrap(), Round::from(commits.round))
```

Comment:
This should delete for every round for the height. Because it's possible the proposal proposed by this node at a particular round was actually passed on, and another node proposed a different value at a higher round.